source/particle/property/velocity_gradient.cc

```diff
@@ -0,0 +1,98 @@
+/*
+  Copyright (C) 2015 - 2025 by the authors of the ASPECT code.
```

Comment:
you lost a lot of empty lines in this license header. Please copy one from an existing file and update the years. Same for the header file.